new_platforms/vsextension/ProjectWizard/ImportEnclaveCommand.cs

```diff
@@ -333,25 +370,46 @@ private async void Execute(object sender, EventArgs e)
-                    var vcProject = project.Object as VCProject;
-                    foreach (var config in vcProject.Configurations)
+                    foreach (VCConfiguration config in vcProject.Configurations)
                     {
+                        var config3 = config as VCConfiguration3;
                         string name = config.Name;
                         if (name.Contains("ARM"))
                         {
                             var clRule = config.Rules.Item("CL") as IVCRulePropertyStorage;
                             string value = clRule.GetUnevaluatedPropertyValue("PreprocessorDefinitions");
                             clRule.SetPropertyValue("PreprocessorDefinitions", "_ARM_;" + value);
 
-                            var config3 = config as VCConfiguration3;
-                            config3.SetPropertyValue("Configuration", true, "WindowsSDKDesktopARMSupport", "true");
-                            config3.SetPropertyValue("Configuration", true, "WindowsSDKDesktopARM64Support", "true");
+                            if (isWindows) {
+                                // Enable compiling Win32 for ARM.
+                                config3.SetPropertyValue("Configuration", true, "WindowsSDKDesktopARMSupport", "true");
+                                config3.SetPropertyValue("Configuration", true, "WindowsSDKDesktopARM64Support", "true");
+                            }
+                        }
+
+                        if (!isWindows && name.Contains("Debug"))
+                        {
+                            // GCC has no preprocessor define for debug mode, but the generated host file
+                            // expects _DEBUG, so set it here.
+                            var clRule = config.Rules.Item("CL") as IVCRulePropertyStorage;
+                            string value = clRule.GetUnevaluatedPropertyValue("PreprocessorDefinitions");
+                            clRule.SetPropertyValue("PreprocessorDefinitions", "_DEBUG;" + value);
                         }
 
-                        // Change OutDir to $(SolutionDir)bin\$(Platform)\$(Configuration)\
-                        // so it's the same as the enclave.
-                        config.OutputDirectory = "$(SolutionDir)bin\\$(Platform)\\$(Configuration)\\";
+                        if (isWindows)
+                        {
+                            // Change OutDir to $(SolutionDir)bin\$(Platform)\$(Configuration)\
+                            // so it's the same as the enclave.
+                            config.OutputDirectory = "$(SolutionDir)bin\\$(Platform)\\$(Configuration)\\";
+                        }
+                        else
+                        {
+                            // Add a post-build event to copy the enclave binary to the existing OutDir.
+                            string cmd = "cp $(RemoteRootDir)/" + baseName + "/bin/$(Platform)/$(Configuration)/" + baseName + ".signed $(RemoteOutDir)";
```

Comment:
Can this be added to the targets file instead?

Reply:
Not easily no.  That's because it has to reference a specific other project.  I'd have to add something else here that the targets file could then use, which is more complicated than just doing it here.